### PR TITLE
Bootstrap nodes sequentially

### PR DESCRIPTION
**Pull request by margdoc**
The concurrent cluster bootstrap is not safe, thus https://github.com/scylladb/scylladb/commit/810b424a8c3612a0a3419cd556839596e381829a was committed. It breaks the cluster setup in jepsen tests, so nodes have to be bootstrapped sequentially.

### CODE REVIEW

**Comment by margdoc** [comment]
@kostja  @kbr- please review

File: src/scylla/db.clj [hunk 1]
```diff
@@ -439,6 +445,18 @@
...
 
+(defn wait-for-previous [test node]
+  (let [nodes (:nodes test)
+        fst (first nodes)]
+    (when (not= node fst)
+        (loop [[previous & tail] nodes]
+            (if (= node (first tail))
+                (do
+                  (info "Waiting for" previous)
+                  (sc/close! (sc/await-open test previous))
```
Comment (kostja) [inline src/scylla/db.clj:447]: qWhat does this do exactly? How do you wait tha tthe previous node is started? The check got to be more sophisticated - I recommend grepping the log for 'serving', or opening a CQL connection and executing a test query, or both. You can't be too cautious here, we have had lots of flakiness earlier when we declared half-started nodes ready.

Comment (kbr-) [reply src/scylla/db.clj:447]: > opening a CQL connection and executing a test query, or both

I think that's exactly what `sc/await-open` does:
```clojure
(defn await-open
  "Blocks until a connection is available, then returns that connection."
  [test node]
  (dt/with-retry [tries 20]
    (let [c (open test node)]
      (alia/execute (:session c)
                    (hayt/->raw (hayt/select :system.peers)))
      c)
    (catch NoHostAvailableException e
      (when (zero? tries)
        (throw+ {:type :jepsen.db/setup-failed
                 :msg  :await-open-timeout
                 :node node}))
      (info node "not yet available, retrying")
      (Thread/sleep await-open-interval)
      (retry (dec tries)))))

```

File: src/scylla/db.clj [hunk 2]
```diff
@@ -439,6 +445,18 @@
...
+                  (info previous "is available"))
+                (recur tail))))))
```
Comment (kbr-) [inline src/scylla/db.clj:449]: It looks like you reimplemented `map` here.
Although for this use case, `run!` seems like a better candidate:
https://clojuredocs.org/clojure.core/run!

File: src/scylla/db.clj [hunk 3]
```diff
@@ -330,7 +330,10 @@
   "Returns a comma-separated string of seed nodes for boostrap.
   Return all nodes to speed up bootstrap."
   [test]
-  (->> (:nodes test)
+  (->> test
+       :nodes
+       first
+       list
```
Comment (kbr-) [inline src/scylla/db.clj]: This is super weird, you're taking the first element of `(:nodes test)`, then making a single-element list of it, then resolving the address and `join`ing the list with `"`...

How about
```
(->> test :nodes first dns-resolve str)
```

Also, the doc string needs to be updated.

File: src/scylla/db.clj [hunk 4]
```diff
@@ -340,7 +343,10 @@
    exclude the node being added from the list of seeds to ensure
    the new node is streamed to before it begins serving reads and writes."
   [test node]
-  (->> (disj (into #{} (:nodes test)) node)
+  (->> test
+       :nodes
+       first
+       list
        (map dns-resolve)
        (str/join ",")))
```
Comment (kbr-) [inline src/scylla/db.clj]: Same comment as above.

**Review by kbr- — CHANGES_REQUESTED** [review]

File: src/scylla/db.clj (hunk 3 above)
Comment (kbr-) [reply src/scylla/db.clj]: Also, piping `test` to `:nodes` using `->>` is kinda awkward, I think you should stick to `(:nodes test)`.

The `->>` macro is good for expressing a sequence of transformations performed on some data structure. But `:nodes` is not a "transformation" performed on `test` - well formally it is, but it's weird for the reader of the code to look at this in this manner.

File: src/scylla/db.clj (hunk 2 above)
Comment (kbr-) [reply src/scylla/db.clj:449]: Sorry, you actually didn't implement `map`.
You want to process an element IF the subsequent element satisfies a predicate. (if it's `node`)
So nevermind.

**Comment by margdoc** [comment]
v1:
* updated comments
* added comment to ```wait-for-previous```
* fixed piping

File: src/scylla/db.clj (hunk 4 above)
Comment (margdoc) [reply src/scylla/db.clj]: Done.

File: src/scylla/db.clj (hunk 3 above)
Comment (margdoc) [reply src/scylla/db.clj]: Done.

File: src/scylla/db.clj (hunk 1 above)
Comment (margdoc) [reply src/scylla/db.clj:447]: As Kamil said, ```sc/await-open``` waits by sending test queries.

**Review by kbr- — APPROVED** [review]

**Review by kostja — APPROVED** [review]

**Pull request merged**